### PR TITLE
Update to Classes

### PR DESCRIPTION
**Pull request by jaredgalanis**

### CODE REVIEW

File: app/app.js [hunk 1]
```diff
@@ -1,14 +1,12 @@
-import Application from '@ember/application';
-import Resolver from './resolver';
-import loadInitializers from 'ember-load-initializers';
-import config from './config/environment';
+import Application from '@ember/application'
```
Comment (locks) [inline app/app.js:1]: semicolons were removed in this file

File: app/adapters/application.js [hunk 2]
```diff
@@ -25,7 +24,7 @@ export default DS.JSONAPIAdapter.extend({
     } else if(requestType === 'query') {
       url = [modelName, `${query.path}-${query.version}.json`];
     } else {
-      return this._super(...arguments);
```
Comment (locks) [inline app/adapters/application.js:28]: Doesn't this change the behavior?

Comment (jaredgalanis) [reply app/adapters/application.js:28]: sure would. thanks for catching that :)

File: app/app.js (hunk 1 above)
Comment (jaredgalanis) [reply app/app.js:1]: copy pasta

**Comment by jaredgalanis** [comment]
> I tried doing a review commit by commit but there seem to be multiple things changing in each one so I was getting confused. If you could split non-`class` changes into separate PRs I can review and get them in more quickly!

@locks  I backed out changes that were not directly related to the class conversion where possible. I'll open another PR that moves the processing of results which I think will get rid of some visual shifting that's happening while results are processing / async loading is occurring.

Note, I have not figured out how to deal with converting the `search-input` component. There's a strange thing with dynamically importing algolia in that file if I convert it to a class. `ember-auto-import` cannot find it on the registry. I can try to work more on this later today or tomorrow, but I'm also okay with getting this work merged and then finding a solution for that issue.

I suppose we may also want to merge your #593 first and then rebase this work on top of that.

**Pull request merged**